datajoint/base_relation.py

```diff
@@ -70,8 +71,14 @@ def declare(self):
         Loads the table heading. If the table is not declared, use self.definition to declare
         """
         if not self.is_declared:
-            self.connection.query(
-                declare(self.full_table_name, self.definition, self._context))
+            if self.definition is not None and self.definition is not Ellipsis:
+                self.connection.query(
+                    declare(self.full_table_name, self.definition, self._context))
+            else:
+                name = self.__class__.__name__
+                warn("""Table for %s is not declared and definition is not defined.
```

Comment:
Why not throw an error? I think it will be more helpful if it can thrown an error for debug purpose. If any code uses the table depending on its functionality, warning will lead to a hard to debug code.